### PR TITLE
feat: refactor route unit test

### PR DESCRIPTION
**Pull request by starsz**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

#1161 
___
### New feature or improvement
- Describe the details and related test reports.

Remove the etcd dependency

### CODE REVIEW

**Comment by membphis** [comment]
ping @starsz

**Comment by starsz** [comment]
> ping @starsz

Sorry, I am too busy to deal with it on those days.
I will update it when I am free.

**Comment by juzhiyuan** [comment]
sync with master?

**Review by membphis — COMMENTED** [review]
big PR, can we split this PR to small PRs?

File: api/internal/handler/route/route_test.go [hunk 1]
```diff
@@ -14,1743 +14,1421 @@
...
+	upstreamRet  interface{}
+	upstreamErr  error
+}
 
-	handler := &Handler{
-		routeStore:    store.GetStore(store.HubKeyRoute),
-		svcStore:      store.GetStore(store.HubKeyService),
-		upstreamStore: store.GetStore(store.HubKeyUpstream),
-		scriptStore:   store.GetStore(store.HubKeyScript),
+var DagScript = `
```
Comment (nic-chen) [inline api/internal/handler/route/route_test.go:55]: we could put these test data to a file in  dir `testdata`,  this is a issue left over from history

File: api/internal/handler/route/route.go [hunk 2]
```diff
@@ -570,30 +557,26 @@ func (h *Handler) Exist(c droplet.Context) (interface{}, error) {
...
+		},
 		PageSize:   0,
 		PageNumber: 0,
 	})
-
 	if err != nil {
 		return nil, err
 	}
 
-	sort := store.NewSort(nil)
```
Comment (nic-chen) [inline api/internal/handler/route/route.go:585]: maybe we could remove these functions now ? I’m not sure if it’s used anywhere else

File: api/internal/handler/route/route_test.go (hunk 1 above)
Comment (starsz) [reply api/internal/handler/route/route_test.go:55]: OK. I will move the DagScript to `dag_conf.go`.
I think ".go" is easier to use than ".json"

Comment (starsz) [reply api/internal/handler/route/route_test.go:55]: OK. I will move the DagScript to `dag_conf.go`.
I think ".go" is easier to use than ".json"

Comment (starsz) [reply api/internal/handler/route/route_test.go:55]: OK. I will move the DagScript to `dag_conf.go`.
I think ".go" is easier to use than ".json"

Comment (starsz) [reply api/internal/handler/route/route_test.go:55]: No. I don't agree to read a JSON file in the unit test.

File: api/internal/handler/route/route.go (hunk 2 above)
Comment (starsz) [reply api/internal/handler/route/route.go:585]: I had checked this. It was still used in upstream.go now. Maybe next time we can delete them.

**Comment by membphis** [comment]
@starsz do you need any help? the frontend e2e failed

**Comment by membphis** [comment]
> big PR, can we split this PR to small PRs?

ping @starsz

**Comment by juzhiyuan** [comment]
> sync with master?

should sync with the master branch.

**Comment by starsz** [comment]
> > big PR, can we split this PR to small PRs?
> 
> ping @starsz

This can be solved by sync the master branch.

**Review by nic-chen — APPROVED** [review]

File: api/internal/handler/route/route_test.go [hunk 3]
```diff
@@ -14,1743 +14,1421 @@
...
+					"version": "v1",
+				},
+			},
+			serviceRet:  "service",
+			serviceErr:  nil,
+			upstreamRet: "upstream",
+			upstreamErr: nil,
+			scriptRet:   "script",
+			scriptErr:   nil,
+			wantRet:     nil,
```
Comment (Jaycean) [inline api/internal/handler/route/route_test.go:748]: I have a question: 
why the wantret nil returned?
This is not consistent with the interface expectation. 
Normally, it should return data.

Comment (imjoey) [reply api/internal/handler/route/route_test.go:748]: +1. We shall keep the behavior of the mock object same as the real one.

Comment (starsz) [reply api/internal/handler/route/route_test.go:748]: Good capture ! Let me fix this, and check other testcase.

Comment (starsz) [reply api/internal/handler/route/route_test.go]: Fixed. Please review again. @Jaycean @imjoey

**Review by imjoey — APPROVED** [review]
/lgtm

File: api/internal/handler/route/route_test.go (hunk 1 above)
Comment (idbeta) [reply api/internal/handler/route/route_test.go:55]: why? Isn't cleaner after code and data separation ?

Comment (starsz) [reply api/internal/handler/route/route_test.go:55]: Read the e2e `testdata` at the unit test.
It seems strange for me.

**Pull request merged**